libs/components/packages/src/schematics/migrations/update-11/vscode-eslint-setting/vscode-eslint-setting.schematic.spec.ts

```diff
@@ -0,0 +1,261 @@
+import { Tree } from '@angular-devkit/schematics';
+import {
+  SchematicTestRunner,
+  UnitTestTree,
+} from '@angular-devkit/schematics/testing';
+
+describe('vscode-eslint-setting.schematic', () => {
+  const possibleLegacyConfigFiles = [
+    '.eslintrc.js',
+    '.eslintrc.cjs',
+    '.eslintrc.yaml',
+    '.eslintrc.yml',
+    '.eslintrc.json',
+  ];
+
+  possibleLegacyConfigFiles.forEach((possibleLegacyConfigFile) => {
+    it(`should run successfully (config: ${possibleLegacyConfigFile}, existingSettings: false)`, async () => {
+      const collectionPath = require.resolve('../../migration-collection.json');
+      jest.mock('child_process', () => ({
+        execSync: jest.fn(),
+      }));
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+      const tree = new UnitTestTree(Tree.empty());
+      tree.create(possibleLegacyConfigFile, 'TESTING');
+      await runner.runSchematic('vscode-eslint-setting', undefined, tree);
```

Comment:
Most of this boilerplate code could be in a `setupTest` function or `beforeEach`.

Reply:
Done. I originally was going to put the running of the schematic in the function as well. However, the `package.json` tests are a bit custom and the branching logic felt like more than it was worth - so I left the running in the individual tests to that I could modify the tree after the setup function is called.